src/backend/distributed/executor/executor.c

```diff
@@ -530,20 +542,19 @@ CreateDistributedExecution(DistributedPlan *distributedPlan,
 
 
 /*
- * StartDistributedExecution opens connections for distributed execution and
- * assigns each task with shard placements that have previously been modified
- * in the current transaction to the connection that modified them.
+ * StartDistributedExecution sets up the coordinated transaction and 2PC for
+ * the execution whenever necessary. It also keeps track of parallel relation
+ * accesses to enforce restrictions that arise due to foreign keys to reference
+ * tables.
  */
 void
 StartDistributedExecution(DistributedExecution *execution)
 {
-	DistributedPlan *distributedPlan = execution->plan;
-	Job *job = distributedPlan->workerJob;
-	List *taskList = job->taskList;
+	List *taskList = execution->tasksToExecute;
 
 	if (MultiShardCommitProtocol != COMMIT_PROTOCOL_BARE)
 	{
-		if (DistributedStatementRequiresRollback(distributedPlan))
+		if (DistributedStatementRequiresRollback(execution->operation, taskList))
```

Comment:
Maybe I should just rename this function to `DistributedExecutionRequiresRollback()` and pass `execution` itself?

Reply:
seems fine